File: game-server/src/main/java/net/dodian/runescape/util/SpawnDirection.java
package net.dodian.runescape.util;

import java.util.Arrays;

public enum SpawnDirection {
    NORTH_WEST(0),
    NORTH(1),
    NORTH_EAST(2),
    WEST(3),
    EAST(4),
    SOUTH_WEST(5),
    SOUTH(6),
    SOUTH_EAST(7);

    private final int id;

    SpawnDirection(int id) {
        this.id = id;
    }

    public int getId() {
        return id;
    }

    public static SpawnDirection getById(int id) {
        return Arrays.stream(SpawnDirection.values())
                .filter(direction -> direction.getId() == id)
                .findFirst().orElse(SOUTH);
    }
}

File: common/game-models/src/main/kotlin/net/dodian/models/SpawnDirection.kt
package net.dodian.models

import java.util.*

enum class SpawnDirection(val id: Int) {
    NORTH_WEST(0),
    NORTH(1),
    NORTH_EAST(2),
    WEST(3),
    EAST(4),
    SOUTH_WEST(5),
    SOUTH(6),
    SOUTH_EAST(7);

    companion object {
        fun getById(id: Int): SpawnDirection {
            return Arrays.stream(values())
                .filter { direction: SpawnDirection -> direction.id == id }
                .findFirst().orElse(SOUTH)
        }
    }
}